src/datanode/src/datanode.rs

```diff
@@ -81,6 +82,7 @@ pub struct Datanode {
     greptimedb_telemetry_task: Arc<GreptimeDBTelemetryTask>,
     leases_notifier: Option<Arc<Notify>>,
     plugins: Plugins,
+    system_metric_task: Option<SystemMetricTask>,
```

Comment:
A question: i wonder if we can implement this task as a `Plugin`. And we can refactor the telemetry task as a plugin too. I don't like explicitly adding too many tasks into `Datanode` or `Frontend`.

Reply:
I'm not sure if the current `Plugin` interface is sufficient for such tasks. Let's continue this topic in another issue and do this refactor later as a follow-up task

Reply:
Issues here https://github.com/GreptimeTeam/greptimedb/issues/2972, cc @shuiyisong who has given the same suggestion